src/cypress-api/index.ts

```diff
@@ -0,0 +1,65 @@
+import type { elementNode } from '@chromaui/rrweb-snapshot';
+import { writeTestResult } from '../write-archive';
+import { SourceMapper } from '../utils/source-mapper';
+import type { ChromaticStorybookParameters } from '../types';
+import type { ResourceArchive } from '../resource-archive';
+
+interface ArchiveParams {
+  testTitle: string;
+  domSnapshots: elementNode[];
+  resourceArchive: ResourceArchive;
+  chromaticStorybookParams: ChromaticStorybookParameters;
+}
+
+const doArchive = async ({
+  testTitle,
+  domSnapshots,
+  resourceArchive,
+  chromaticStorybookParams,
+}: ArchiveParams) => {
+  let sourceMap = null;
+
+  if (domSnapshots.length > 0) {
+    // shortens file names in the last snapshot (which is the automatic one)
+    const sourceMapper = new SourceMapper(domSnapshots[domSnapshots.length - 1]);
+    sourceMap = sourceMapper.shortenFileNamesLongerThan(250).build();
+  }
+
+  const bufferedArchiveList = Object.entries(resourceArchive).map(([key, value]) => {
+    return [
+      key,
+      {
+        ...value,
+        // we can't use Buffer in the browser (when we collect the responses)
+        // so we go through one by one here and bufferize them
+        // @ts-expect-error will fix when Cypress has its own package
+        body: Buffer.from(value.body, 'utf8'),
+      },
+    ];
+  });
+
+  const allSnapshots = Object.fromEntries(
+    domSnapshots.map((item, index) => [`Snapshot #${index + 1}`, Buffer.from(JSON.stringify(item))])
+  );
+
+  await writeTestResult(
+    // @ts-expect-error will fix when Cypress has its own package
+    {
+      title: testTitle,
+      // doesn't matter what value we put here, as long as it's a subdirectory of where we want this to actually go
+      // TODO: change so we don't have to do this trickery
+      outputDir: './some',
+    },
+    allSnapshots,
+    Object.fromEntries(bufferedArchiveList),
+    { ...chromaticStorybookParams, viewport: { width: 500, height: 500 } },
+    sourceMap
+  );
+};
+
+export const archiveCypress = (stuff: ArchiveParams): null => {
+  doArchive(stuff);
+
+  // Cypress tasks must return a value or null: https://docs.cypress.io/api/commands/task#Usage
+  return null;
+};
```

Comment:
This is the actual Cypress task that enables passing the archives and snapshots from browser to server.

Reply:
We aren't awaiting `doArchive`. Is that a problem? What if the process exits before we are finished?

Reply:
Could we use [writeFile](https://docs.cypress.io/api/commands/writefile) to do away with the need for the node server callback?

Reply:
I changed to awaiting `doArchive`. @tevanoff I put in a [ticket](https://linear.app/chromaui/issue/AP-3823/write-files-from-cypress-directly) for writing to files from the Cypress side for us to consider in the future.